### DISCUSSION TITLE
New (Rust) dependencies make it impossible to package jsonschema for a number of architectures

### DISCUSSION THREAD

**Discussion by mgorny** [python-jsonschema/jsonschema#1118] [General]
I'm not hoping for much but I just would like to let you know that through making jsonschema depend on Rust (via `rpds-py`, used directly and indirectly), you're effectively making it impossible to use it on a number of systems where Rust doesn't work (this includes less common architectures as well as older versions of more common architectures such as ARM).

I can't at this point tell you how much damage this causes because I'm going to need to find a few hours just to assess it, given how many indirect dependencies `jsonschema` has at this point. I'm somewhat reluctant to even start because 1) I'm completely swamped with work, and 2) I have a growing suspicion that sooner or later I'm going to discover that we can't package `setuptools` anymore and we effectively have to kill Python on all these architectures, so I'll effectively spend a few hours trying to salvage something only to discover that all hope is gone.

**Comment by Julian** [comment]
If it makes things any better, `jsonschema` will shortly also have a pure Python fallback intended to increase (or really recoup) performance on PyPy. That may help you as well, though I definitely won't be able to put any effort into doing so myself.

**Reply by mgorny** [reply]
Yes, pure Python fallback would certainly be helpful. Also note that poetry-core is now affected since it can't vendor new versions of jsonschema while they require Rust: https://github.com/python-poetry/poetry-core/commit/22e69d51b82653c0419eebfca3eb01e8f96faa5d#diff-96dcc6a8c78ceced3d29c2d4702dd26681ef29aec4fb454071f4730116540473

**Comment by jbyibm** [comment]
I found this discussion after creating an issue here:
* https://github.com/python-jsonschema/jsonschema/issues/1143

Maybe Rust is not available also on Z Linux (s390)
